### PR TITLE
Rename object group with f2

### PR DESCRIPTION
**Pull request by AlexandreSi**
To have the same behavior between objects list and groups list.

@danired we didn't think about this: users wouldn't understand that this feature doesn't work in the groups list.

### CODE REVIEW

**Comment by danired** [comment]
Haha that's good!

I really thought about this feature but I saw that when clicking a Group name, the group is not highlighted as selected, so I thought it would be a little harder and I put the task in the Backlog.

One question: When both a Object and a Group are selected and you press F2, which one begins the rename process?

**Comment by AlexandreSi** [comment]
You can see in the

**Comment by danired** [comment]
> You can see in the PR that I added a state variable to the SceneEditor component that stores which editor is active...

Good one! You made a great work.

File: newIDE/app/src/SceneEditor/Toolbar.js [hunk 1]
```diff
@@ -65,8 +64,7 @@ const Toolbar = (props: Props) => {
         canDeleteSelection={
           props.instancesSelection.getSelectedInstances().length !== 0
         }
-        canRenameObject={props.canRenameObject}
```
Comment (ClementPasteau) [inline newIDE/app/src/SceneEditor/Toolbar.js:68]: is this not relevant anymore to be able to disable the action?

File: newIDE/app/src/SceneEditor/ToolbarCommands.js [hunk 2]
```diff
@@ -65,8 +64,8 @@ const ToolbarCommands = (props: Props) => {
     handler: props.setupGrid,
   });
 
-  useCommand('RENAME_SCENE_OBJECT', props.canRenameObject, {
-    handler: props.onRenameObject,
+  useCommand('RENAME_SCENE_OBJECT', true, {
```
Comment (ClementPasteau) [inline newIDE/app/src/SceneEditor/ToolbarCommands.js:67]: command name to update?

File: newIDE/app/src/SceneEditor/index.js [hunk 3]
```diff
@@ -154,6 +154,7 @@ type State = {|
 
   renamedObjectWithContext: ?ObjectWithContext,
   selectedObjectsWithContext: Array<ObjectWithContext>,
+  activeEditor: string | null,
```
Comment (ClementPasteau) [inline newIDE/app/src/SceneEditor/index.js:157]: maybe type this prop with a list of possible editors?

File: newIDE/app/src/UI/SortableVirtualizedItemList/index.js [hunk 4]
```diff
@@ -27,6 +27,7 @@ type Props<Item> = {|
   getItemData?: (Item, index: number) => HTMLDataset,
   isItemBold?: Item => boolean,
   onItemSelected: (?Item) => void,
+  itemEqualityTest?: (Item, Item) => boolean,
```
Comment (ClementPasteau) [inline newIDE/app/src/UI/SortableVirtualizedItemList/index.js:30]: I'd rather call this function with a verb to be consistent: `areItemsEqual()`

File: newIDE/app/src/SceneEditor/Toolbar.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/SceneEditor/Toolbar.js:68]: I could add something but the worst case is that we try to rename an object that is the first element of a list, that would be `undefined` if the list is empty. Both methods handle that case. Do you think I should add this?

File: newIDE/app/src/SceneEditor/index.js (hunk 3 above)
Comment (AlexandreSi) [reply newIDE/app/src/SceneEditor/index.js:157]: Good idea!

File: newIDE/app/src/SceneEditor/index.js [hunk 5]
```diff
@@ -718,8 +728,15 @@ export default class SceneEditor extends React.Component<Props, State> {
...
-  _startRenamingSelectedObject = () => {
-    this._onRenameObjectStart(this.state.selectedObjectsWithContext[0]);
+  _startRenamingSelectedObjectOrGroup = () => {
+    if (this.state.activeEditor === 'objects-list') {
+      this._onRenameObjectStart(this.state.selectedObjectsWithContext[0]);
+    } else if (
+      this.state.activeEditor === 'object-groups-list' &&
+      this._objectGroupsList
+    ) {
+      this._objectGroupsList.startEditingSelectedGroup();
```
Comment (4ian) [inline newIDE/app/src/SceneEditor/index.js:738]: this is the thing I'm worried about. I'm unsure if this notion of "activeEditor" is the right one.

Comment (AlexandreSi) [reply newIDE/app/src/SceneEditor/index.js:738]: I couldn't find something similar in react-mosaic API.

**Pull request closed without merging**